### ISSUE TITLE
Support for "enforce" when using merge.smart

### ISSUE THREAD

**Issue by patrick91** [survivejs/webpack-merge#65]
Hi :)

Since webpack 2, preLoaders have been removed in favour of the "enforce" rule, see: https://webpack.js.org/configuration/module/#rule-enforce

It seems to me that webpack merge doesn't correctly support it. It should not merge rules with different enforce rules.

Sample here:

```js
var merge = require('webpack-merge')

const a = {
  module: {
    rules: [
      {
        test: /\.vue$/,
        loader: 'eslint-loader',
        enforce: 'pre',
        exclude: /node_modules/
      },
    ]
  }
}

const b = {
  module: {
    rules: [
      {
        test: /\.vue$/,
        loader: 'vue-loader',
      },
    ]
  }
}

const merged = merge.smart(a, b)

function stringifyFilter(key,value) {
    if (value instanceof RegExp) {
        return value.toString();
    }

    return value;
}

console.log(JSON.stringify(merged, stringifyFilter, "    "));
```

Output:
```json
{
    "module": {
        "rules": [
            {
                "test": "/\\.vue$/",
                "loader": "vue-loader",
                "enforce": "pre",
                "exclude": "/node_modules/"
            }
        ]
    }
}
```

**Comment by bebraw** [comment]
Yeah. That's a good point. It's going to take some extra code to support this special case. Thanks for bringing this up. 👍 

I've been thinking of dropping the smart variant for issues like this. Lots of corner cases.

**Comment by patrick91** [comment]
@bebraw looks like it is already supported: https://github.com/survivejs/webpack-merge/blob/master/src/join-arrays-smart.js#L9

There's just a small issue, I think it should just check if the rules.enforce are difference, without checking if the enforce property exists

**Comment by bebraw** [comment]
Fixed in `2.6.1`. You were right, it was a simple change to make.
